packages/http-client-js/src/components/client-context/client-context-factory.tsx

```diff
@@ -75,66 +76,112 @@ interface ClientFactoryArgumentsProps {
 function ClientFactoryArguments(props: ClientFactoryArgumentsProps) {
   const params = [<>{props.endpointRef},</>];
 
-  if (props.credentialsRef) {
-    params.push(<>{props.credentialsRef},</>);
-  }
-
   return [
     ...params,
-    <ClientOptionsExpression>
-      <CredentialOptions client={props.client} />
-    </ClientOptionsExpression>,
+    props.credentialsRef ? (
+      <ClientOptionsExpression>
+          {props.credentialsRef}
+          <AuthSchemeOptions client={props.client} />
+      </ClientOptionsExpression>
+    ) : (
+      <ClientOptionsExpression />
+    ),
   ];
 }
 
-interface CredentialOptionsProps {
+interface AuthSchemeOptionsProps {
   client: cl.Client;
 }
 
-function CredentialOptions(props: CredentialOptionsProps) {
-  const clientCredential = $.client.getAuth(props.client);
-
-  if (clientCredential.schemes.length === 0) {
-    return null;
-  }
-
-  if (clientCredential.schemes.length !== 1) {
-    reportDiagnostic($.program, {
-      code: "multiple-auth-schemes-not-yet-supported",
-      target: props.client.type,
-    });
-  }
-
-  const scheme = clientCredential.schemes[0];
+interface AuthSchemeProps {
+  scheme: HttpAuth;
+  client: cl.Client;
+}
 
-  switch (scheme.type) {
+function AuthScheme(props: AuthSchemeProps) {
+  switch (props.scheme.type) {
     case "http":
-      // Todo: handle scopes?
       return (
-        <ts.ObjectProperty name="credentials">
-          <ts.ObjectExpression>
-            <ts.ObjectProperty name="apiKeyHeaderName" jsValue={"Authorization"} />
-          </ts.ObjectExpression>
-        </ts.ObjectProperty>
+        <ts.ObjectExpression>
+          <ts.ObjectProperty name="kind" jsValue="http" />
```

Comment:
I think you can use <List comma> to wrap the ObjectProperties and it should take care of the commas